### PR TITLE
feat: option to write body facets on save

### PR DESCRIPTION
**Pull request by jonahrb**
## Description
Write body facets into the scdocx on save

## Checklist
- [ ] I have tested my changes locally.
- [ ] I have added necessary documentation or updated existing documentation.
- [ ] I have followed the coding style guidelines of this project.
- [ ] I have added appropriate unit tests.
- [ ] I have reviewed my changes before submitting this pull request.
- [ ] I have linked the issue or issues that are solved to the PR if any.
- [ ] I have assigned this PR to myself.
- [ ] I have added the minimum version decorator to any new backend method implemented.
- [ ] I have made sure that the title of my PR follows [Conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. ``feat: extrude circle to cylinder``)

### CODE REVIEW

**Review by RyanJWard — CHANGES_REQUESTED** [review]
@jonahrb We need to skip these tests for older version and we might need a minimum version right?

File: src/ansys/geometry/core/designer/design.py [hunk 1]
```diff
@@ -351,14 +364,18 @@ def __export_and_download(self, format: DesignFileFormat) -> bytes:
...
+                )
             except Exception:
                 self._grpc_client.log.warning(
                     f"Failed to download the file in {format} format."
                     " Attempting to stream download."
                 )
                 # Attempt to download the file via streaming
-                response = self._grpc_client.services.designs.stream_download_export(format=format)
+                response = self._grpc_client.services.designs.stream_download_export(
+                    format=format, write_body_facets=write_body_facets
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/design.py:377]: Agreed.. the write_body_facets, in case it is True, should be verified against the backend version. If True and backend_version < 26.1 then we need to throw a warning and inform that this option will not be considered since the feature wasn't available prior to 26.1

Comment (RobPasMue) [reply src/ansys/geometry/core/designer/design.py:377]: This comment applies to the three methods. Maybe you want to come up with a common method and reuse it. Something like ``check_write_body_facets_input(write_body_facets, backend_version)``.

Comment (RobPasMue) [reply src/ansys/geometry/core/designer/design.py:377]: I would do something like this in fact:

```py
        # HACK: we should inform the user that NURBS curve sketches are not supported
        # prior to 26R1... and if passed, raise an error
        _nurbs_curves_compatibility(kwargs["backend_version"], request.geometries)
 ```
 
 This is already being done inside the ``_grpc/_services/v0/bodies.py`` file. Could you implement this @jonahrb ?

**Review by RyanJWard — APPROVED** [review]

**Review by RobPasMue — APPROVED** [review]
Awesome, LGTM!
